contracts/kleros/KlerosLiquid.sol

```diff
@@ -0,0 +1,724 @@
+pragma solidity ^0.4.24;
+
+import "kleros-interaction/contracts/standard/arbitration/Arbitrator.sol";
+import "kleros-interaction/contracts/standard/arbitration/Arbitrable.sol";
+import "kleros-interaction/contracts/standard/rng/RNG.sol";
+import { MiniMeTokenERC20 as Pinakion } from "kleros-interaction/contracts/standard/arbitration/ArbitrableTokens/MiniMeTokenERC20.sol";
+import { TokenController } from "minimetoken/contracts/TokenController.sol";
+
+import "../data-structures/SortitionSumTreeFactory.sol";
+
+/**
+ *  @title KlerosLiquid
+ *  @author Enrique Piqueras - <epiquerass@gmail.com>
+ *  @dev The main Kleros contract with dispute resolution logic for the Athena release.
+ */
+contract KlerosLiquid is SortitionSumTreeFactory, TokenController, Arbitrator {
+    /* Enums */
+
+    // General
+    enum Phase {
+      staking, // Stake sum trees can be updated. Pass after `minStakingTime` passes and there is at least one dispute without jurors.
+      generating, // Waiting on random number. Pass as soon as it is ready.
+      drawing // Jurors can be drawn. Pass after all disputes have jurors or `maxDrawingTime` passes.
+    }
+
+    // Dispute
+    enum Period {
+      evidence, // Evidence can be submitted. This is also when drawing has to take place.
+      commit, // Jurors commit a hashed vote. This is skipped if not a hidden court.
+      vote, // Jurors reveal/cast their vote depending on wether the court is hidden or not.
+      appeal, // The dispute can be appealed.
+      execution // Tokens are redistributed and the ruling is executed.
+    }
+
+    /* Structs */
+
+    // General
+    struct Court {
+        uint parent; // The parent court.
+        uint[] children; // List of child courts.
+        uint[] vacantChildrenIndexes; // Stack of vacant slots in the children list.
+        bool hidden; // Wether to use commit and reveal or not.
```

Comment:
`hiddenVotes` ?

Reply:
I don't think so. A 'hidden' court makes it sound like the court is private

Reply:
True. Changed.